test/test_target.py

```diff
@@ -416,6 +416,40 @@ def test_opencl_support_for_bool(ctx_factory):
     np.testing.assert_equal(out, np.tile(np.array([0, 1], dtype=np.bool8), 5))
 
 
+def test_opencl_funcs_resolve_and_run(ctx_factory):
+    from loopy.target.opencl import UNARY_FUNCS, _CL_SIMPLE_MULTI_ARG_FUNCTIONS
+
+    knl = lp.make_kernel(
+        "{:}",
+        "\n".join(f"f[{i}] = {func}(.2341)" for i, func in enumerate(UNARY_FUNCS)),
+    )
+
+    cl_ctx = ctx_factory()
+    _ = knl(cl.CommandQueue(cl_ctx), f=np.zeros(len(UNARY_FUNCS)))
+
+    binary_funcs = {func for func, val in _CL_SIMPLE_MULTI_ARG_FUNCTIONS.items()
+                    if val == 2}
+
+    knl = lp.make_kernel(
+        "{:}",
+        "\n".join(f"f[{i}] = {func}(.2341, .5321)"
+                  for i, func in enumerate(binary_funcs)),
+    )
+
+    _ = knl(cl.CommandQueue(cl_ctx), f=np.zeros(len(binary_funcs)))
+
+    ternary_funcs = {func for func, val in _CL_SIMPLE_MULTI_ARG_FUNCTIONS.items()
+                     if val == 3}
+
+    knl = lp.make_kernel(
+        "{:}",
+        "\n".join(f"f[{i}] = {func}(.2341, .5321, .912)"
+                  for i, func in enumerate(ternary_funcs)),
+    )
+
+    _ = knl(cl.CommandQueue(cl_ctx), f=np.zeros(len(ternary_funcs)))
```

Comment:
Would you be willing to hack together some output checking for these, at least where numpy equivalents exist?

Reply:
It ain't pretty, but: 561c5e0